File: AutoMix Software/AutoMix Data Management/DataBaseExtraction.cpp
// Copyright (C) 2016-2017 LesProjecteurs - All Rights Reserved
// Maxime STEVENOT, Guillaume HANNES, Jordan ERNULT, Louis CARLIER, Pierre GABON
// 
// This file is part of AutoMix Software.
// 
// Unauthorized copying of this file, via any medium is strictly prohibited.
// You should have received a copy of the License along with this program.

#include "stdafx.h"
#include "DataBaseExtraction.h"

using namespace System::Threading::Tasks;
using namespace System::Diagnostics;
using namespace System::IO;
using namespace System;

namespace AutoMixDataManagement {

	DataBaseExtraction::DataBaseExtraction(DataBase^ dataBase)
	{
		_dataBase = dataBase;
	}

	void DataBaseExtraction::extractData(System::ComponentModel::BackgroundWorker^ bw, int nbTracks, System::Threading::CancellationTokenSource^ cts, Track^ track)
	{
		_dataBase->extractData(track);
		bw->ReportProgress((int)100 / nbTracks);
	}
}

File: Automix Data Management/Extraction/DataBaseExtraction.cs
// Copyright (C) 2016 - 2018 LesProjecteurs
// This file is part of Automix Software licensed under MIT License.

using System.ComponentModel;
using System.Threading;
using Automix_Data_Management.Model;
using Automix_Data_Management.Storage;

namespace Automix_Data_Management.Extraction
{
    internal class DataBaseExtraction : IExtraction
    {
        private readonly DataBase _dataBase;

        public DataBaseExtraction(DataBase dataBase)
        {
            _dataBase = dataBase;
        }

        public void ExtractData(BackgroundWorker backgroundWorker, int nbTracks, CancellationTokenSource cancellationTokenSource, Track track)
        {
            _dataBase.ExtractData(track);
            Interlocked.Increment(ref AudioDataExtraction.ExploredTracks);
            backgroundWorker.ReportProgress(500 + 500 * AudioDataExtraction.ExploredTracks / nbTracks);
        }
    }
}